sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -134,22 +110,32 @@ def __init__(self, pre_grouped_coder, post_grouped_coder, windowing):
 
   def append(self, elements_data):
     input_stream = create_InputStream(elements_data)
+    coder_impl = self._pre_grouped_coder.get_impl()
+    key_coder_impl = self._key_coder.get_impl()
+    # TODO(robertwb): We could optimize this even more by using a
+    # window-dropping coder for the data plane.
+    is_trivial_windowing = self._windowing.is_default()
     while input_stream.size() > 0:
-      windowed_key_value = self._pre_grouped_coder.get_impl(
-          ).decode_from_stream(input_stream, True)
-      key = windowed_key_value.value[0]
-      windowed_value = windowed_key_value.with_value(
-          windowed_key_value.value[1])
-      self._table[self._key_coder.encode(key)].append(windowed_value)
+      windowed_key_value = coder_impl.decode_from_stream(input_stream, True)
+      key, value = windowed_key_value.value
+      self._table[key_coder_impl.encode(key)].append(
+          value if is_trivial_windowing
+          else windowed_key_value.with_value(value))
 
   def __iter__(self):
     output_stream = create_OutputStream()
-    trigger_driver = trigger.create_trigger_driver(self._windowing, True)
+    if self._windowing.is_default():
+      globally_window = GlobalWindows.windowed_value(None).with_value
```

Comment:
<!--new_thread; commit:012f8b80e98468e9e26965cf89cc2cc40b6c5662; resolved:0-->
Did you mean to call this global_window?

Reply:
This is a verb, it's the function that takes a value and globally windows it.